directories/src/redux/epics/node-epics.js

```diff
@@ -143,9 +147,34 @@ const requestGenesisHashEpic = (action$, store) => {
             brokerNode.completeGenesisHashPoW(txid, trytesArray[0])
           ).map(({ data }) => {
             const { purchase: genesisHash, numberOfChunks } = data;
-            return nodeActions.addGenesisHash({ genesisHash, numberOfChunks });
-          })
+            return { genesisHash, numberOfChunks };
+          }).catch(error => {
+            console.log("GENESIS HASH COMPLETE ERROR", error);
+            return Observable.empty();
+          });
         )
+        .mergeMap(({ genesisHash, numberOfChunks }) => {
+          const datamap = Datamap.generate(genesisHash, numberOfChunks);
+          const addresses = _.values(datamap);
+          const randomNumber = AppUtils.randomNumber(addresses.length);
+          const randomSectorAddresses = addresses.slice(randomNumber);
+          return Observable.fromPromise(
+             iota.findTransactions(randomSectorAddresses)
+          ).map(({ transaction }) => {
+            const { bundles, addresses, tags, approvees } = transaction;
+            return { bundles, addreses, tags, approvees, genesisHash, numberOfChunks };
+          });
+        })
+        .mergeMap(({ bundles, addreses, tags, approvees, genesisHash, numberOfChunks}) => {
+          return Observable.fromPromise(
+             iota.getTrytes(bundles)
```

Comment:
You'll be calling getTrytes on the hashes.  Not sure where bundles is coming from here.

Reply:
What I mean is that findTransactions, as far as I know, does not return an object with a "bundles" property.  It just returns an array of hashes.  What I'm saying is that I think there's a problem with lines 162 - 165, because what you are expecting from findTransactions is not what you will actually get.